daemon/network.go

```diff
@@ -866,20 +866,18 @@ func buildCreateEndpointOptions(c *container.Container, n *libnetwork.Network, e
 		createOptions = append(createOptions, libnetwork.CreateOptionDisableResolution())
 	}
 
-	// configs that are applicable only for the endpoint in the network
-	// to which container was connected to on docker run.
-	// Ideally all these network-specific endpoint configurations must be moved under
-	// container.NetworkSettings.Networks[n.Name()]
-	if nwName == c.HostConfig.NetworkMode.NetworkName() || (nwName == defaultNetName && c.HostConfig.NetworkMode.IsDefault()) {
-		if c.Config.MacAddress != "" {
```

Comment:
So the `c.Config.MacAddress` option is a "global" option, and applies to the container's main / primary network, and not a per-network options.

Does this change means that it applies to "any network" attached to the container? What happens if the container would be attached to multiple networks; do all endpoints get the same MacAddress?